nilearn/maskers/surface_labels_masker.py

```diff
@@ -222,19 +249,44 @@ def fit(self, img=None, y=None):
                 "masker = SurfaceLabelsMasker(labels_img=labels_img)"
             )
 
+        if self.labels and self.lut is not None:
+            raise ValueError(
+                "Pass either labels or a lookup table (lut) to the masker, "
+                "but not both."
+            )
+
+        self._shelving = False
+
         all_labels = set(self._labels_data.ravel())
         all_labels.discard(self.background_label)
         self._labels_ = list(all_labels)
 
         self.n_elements_ = len(self._labels_)
 
-        if self.labels is None:
-            self.label_names_ = [str(label) for label in self._labels_]
+        # generate a look up table if one was not provided
+        if self.lut is not None:
+            if isinstance(self.lut, (str, Path)):
+                lut = pd.read_csv(self.lut, sep="\t")
+            else:
+                lut = self.lut
```

Comment:
Can you make sure to test that ?

Reply:
now tested with a dafaframe and path to a TSV

Reply:
actually I decided to use pd.read_table for robustness in case users pass a CSV